cpp/bmi_cpp_extension.cc

```diff
@@ -268,9 +175,11 @@ void BmiCppExtension::GetGridNodesPerFace(int grid, int *dest)
     memcpy(dest, x.data(), x.size()*sizeof(double));
 }
 
-char BmiCppExtension::FindType(const std::string& varname) const
+char BmiCppExtension::FindType(const std::string varname) const
 {
-    std::string vartype = this->GetVarType(varname);
+    // TODO unable to call GetVarType, hardcoding for now
```

Comment:
char varchar[BMI_MAX_TYPE_NAME];
this->GetVarType(varname, varchar);
std::string vartype=std::string(varchar);

Reply:
Thats the C version, in C++ the BMI class defines `virtual std::string GetVarType(std::string name)` so I am confused, because thats what I am giving and recieveing